assets/app/scripts/controllers/overview.js

```diff
@@ -40,22 +53,156 @@ angular.module('openshiftConsole')
       LabelFilter.addLabelSuggestionsFromResources($scope.unfilteredServices, $scope.labelSuggestions);
       LabelFilter.setLabelSuggestions($scope.labelSuggestions);
       $scope.services = LabelFilter.getLabelSelector().select($scope.unfilteredServices);
-      podsByServiceByLabel();
+
+      // Order is important here since podRelationships expects deploymentsByServiceByDeploymentConfig to be up to date
+      deploymentsByService();
+      deploymentConfigsByService();
+      podRelationships();
+
       $scope.emptyMessage = "No services to show";
       updateFilterWarning();
       console.log("services (list)", $scope.services);
     }));
 
-    var podsByServiceByLabel = function() {
-      $scope.podsByServiceByLabel = {};
-      angular.forEach($scope.services, function(service, name) {
-        var ls = new LabelSelector(service.spec.selector);
-        var servicePods = ls.select($scope.pods)
-        $scope.podsByServiceByLabel[name]  =  {};
-        DataService.objectsByAttribute(servicePods, "metadata.labels", $scope.podsByServiceByLabel[name], null, "metadata.name");
+    // Expects deploymentsByServiceByDeploymentConfig to be up to date
+    var podRelationships = function() {
+      $scope.monopodsByService = {"": {}};
+      $scope.podsByService = {};
+      $scope.podsByDeployment = {};
+
+      // Initialize all the label selectors upfront
+      var depSelectors = {};
+      angular.forEach($scope.deployments, function(deployment, depName){
+        depSelectors[depName] = new LabelSelector(deployment.spec.selector);
+        $scope.podsByDeployment[depName] = {};
+      });
+      var svcSelectors = {};
+      angular.forEach($scope.unfilteredServices, function(service, svcName){
+        svcSelectors[svcName] = new LabelSelector(service.spec.selector);
+        $scope.podsByService[svcName] = {};
+      });      
+
+
+      angular.forEach($scope.pods, function(pod, name){
+        var deployments = [];
+        var services = [];
+        angular.forEach($scope.deployments, function(deployment, depName){
+          var ls = depSelectors[depName];
+          if (ls.matches(pod)) {            
+            deployments.push(depName);
+            $scope.podsByDeployment[depName][name] = pod;
+          }
+        });
+        angular.forEach($scope.unfilteredServices, function(service, svcName){
+          var ls = svcSelectors[svcName];
+          if (ls.matches(pod)) {
+            services.push(svcName);
+            $scope.podsByService[svcName][name] = pod;
+
+            var isInDepInSvc = false;
+            // Not using angular.forEach so that we can break out early
+            for (var i = 0; !isInDepInSvc && i < deployments.length; i++) {
```

Comment:
I'd go back to angular.forEach for clarity... `deployments` is unlikely to be longer than 1